lib/tokens/token.js

```diff
@@ -41,6 +41,18 @@ module.exports = function (log, random, P, hkdf, Bundle, error) {
     this.uid = details.uid || null
     this.lifetime = details.lifetime || Infinity
     this.createdAt = optionallyOverrideCreatedAt(details.createdAt)
+    if (details.createdAt) {
+      // We don't expect this to be set outside the tests so log an error
+      // if we do encounter it, to help debug what's going on.
```

Comment:
I'm not sure this comment is accurate - we expect it to be set when reading an existing token from the DB anf using `Token.createTokenFromHexData`, but not when creating a fresh token from scratch using `Token.createNewToken`.  It might make sense to move this check into those two methods instead.